src/runtime/recipe/particle.ts

```diff
@@ -164,17 +178,25 @@ export class Particle {
   set spec(spec) {
     this._spec = spec;
     for (const connectionName of spec.connectionMap.keys()) {
-      const speccedConnection = spec.connectionMap.get(connectionName);
       let connection = this.connections[connectionName];
+      const speccedConnection = spec.connectionMap.get(connectionName);
       if (connection == undefined) {
-        connection = this.addConnectionName(connectionName);
+        if (speccedConnection.type instanceof InterfaceType) {
+          // NOTE: At cloning, all type-variables change. Hence handle / handle-connections created in
+          // different planning generations cannot connect via interface.
```

Comment:
```
interface MyInterface
  in ~a thing

particle MyParticle
  in [~a] list
  host MyInterface hostedParticle

recipe
  ParticleA
```
Today the `list` and the `hostedParticle` handle connections are created at the same time (during manifest parsing and list's `~a` is the same type variable as hostedParticle's interface handle's `~a`., meaning that when one of them is being resolved (specifically in 
https://github.com/PolymerLabs/arcs/blob/dc24d9931e976d86e897871ad8902be726b5030f/src/runtime/recipe/recipe-util.ts#L362 called from https://github.com/PolymerLabs/arcs/blob/8d1f6ba731ba1be34195da7512b98a1ff5817ff1/src/runtime/strategies/find-hosted-particle.ts#L40 ) the other is being resolved to the same type.

With this change, if `list` connection is created in earlier generation, then the recipe is cloned, then the `hostedParticle` connection is created in a different generation, the `~a`s do not reference the same type variable and resolving one doesn't affect the other.

Now that I wrote it, i finally realized that this will apply true to any 2 handle connections, no matter interface or not, so this change is not good. 
Please advise.

Reply:
So we'll need to make sure that handle connection construction picks up on the existing scoped type variables if they've already been constructed.

There's a (currently ephemeral) name map that ties scoped variables together. I guess we'll need to expose this at the particle level and reuse it?

I can take a look into the code changes necessary for this if you want. It's probably a change that could go through first.

Reply:
yes separate change is best, because this pr is already quite big. 
and yes, would be great if you could take a look.  if you don't have time to actually implement it, please, just point me in the right direction and i'll work on this when i get back. Thanks!

Reply:
i see that you didn't have time to address this yet, but i think i know what you meant as the solution. i would prefer to do it in a follow up pr though, if that's ok with you.